File: Medium/2025-11-21-1930-Unique-Length-3-Palindromic-Subsequences/solution.js
var countPalindromicSubsequences = function(s) {
    const seen = new Set();
    const n = s.length;

    for (let i = 0; i < n; i++) {
        for (let j = i + 1; j < n; j++) {
            if (s[i] === s[j]) {
                for (let k = 0; k < n; k++) {
                    if (k !== i && k !== j && s[k] === s[i]) {
                        seen.add(s[i] + s[j] + s[i]);
                    }
                }
            }
        }
    }

    return seen.size;
};

File: Medium/2025-11-21-1930-Unique-Length-3-Palindromic-Subsequences/solution.py
class Solution:
    def uniquePalindromicSubseq(self, s: str) -> int:
        seen = set()
        last = {}
        for i, char in enumerate(s):
            if char in last:
                for j in last[char]:
                    seen.add(char + s[j] + char)
            if char not in last:
                last[char] = []
            last[char].append(i)
        return len(seen)

# Example test cases
sol = Solution()
print(sol.uniquePalindromicSubseq("aabca"))  # Output: 3
print(sol.uniquePalindromicSubseq("adc"))    # Output: 0
print(sol.uniquePalindromicSubseq("bbcbaba")) # Output: 4